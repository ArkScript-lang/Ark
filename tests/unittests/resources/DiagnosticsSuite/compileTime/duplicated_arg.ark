($ -> (arg fn ...fn) {
    ($if (> (len fn) 0)
        (-> (fn1 arg) ...fn)
        (fn1 arg))})

(let filename "a")

(let f1 (fun (data) {
    (+ data "000")}))
(let f2 (fun (data) {
    (+ data "002")}))
(let f3 (fun (data) {
    (+ data "003")}))
(let f4 (fun (data) {
    (+ data "004")}))

(print (-> filename f1 f2 f3 f4))
