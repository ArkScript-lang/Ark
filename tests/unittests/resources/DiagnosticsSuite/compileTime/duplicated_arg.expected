At . @ 1:15
    1 | ($ -> (arg fn ...fn) {
      |               ^~~~~
    2 |     ($if (> (len fn) 0)
    3 |         (-> (fn1 arg) ...fn)
        Argument names must be unique, can not reuse `fn'
